src/gui/qgsmapcanvas.cpp

```diff
@@ -202,6 +202,7 @@ QgsMapCanvas::QgsMapCanvas( QWidget * parent, const char *name )
   mCurrentLayer = NULL;
   mMapOverview = NULL;
   mMapTool = NULL;
+  mOldCursor = cursor();
```

Comment:
Everyone API that changes the cursor or everyone other element has to
restore it in previous state after finishing using it. This means that when
a MapTool is activated and changes the cursor, the cursor has to be
restored to its previous shape when the tool is deactivated.

In the old code was:

{

    mMapTool->deactivate();

    mMapTool = NULL;

    emit mapToolSet( NULL );

    emit mapToolSet( NULL, mMapTool );

    setCursor( Qt::ArrowCursor );

}

When the tool is deactivated the cursor shape is changed to Arrow despite
of its previous shape and it is done after emitting of signal that MapTool
is deactivated!

This means that if you catch the signal outside of mapcanvas and change the
cursor shape, the cursor shape will be changed to Arrow after that.

On Tue, Jul 28, 2015 at 11:31 AM, Matthias Kuhn <notifications@github.com>
wrote:

> In src/gui/qgsmapcanvas.cpp
> <https://github.com/qgis/QGIS/pull/2224#discussion_r35625013>:
>
> > @@ -202,6 +202,7 @@ QgsMapCanvas::QgsMapCanvas( QWidget * parent, const char *name )
> >    mCurrentLayer = NULL;
> >    mMapOverview = NULL;
> >    mMapTool = NULL;
> > +  mOldCursor = cursor();
>
> What is that for?
> Normally the cursor state should be controlled by matching
> QApplication::setOverrideCursor and QApplication::restoreOverrideCursor
> calls.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/2224/files#r35625013>.
>